libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -800,6 +780,36 @@ void ContractLevelChecker::checkOverrideList(FunctionMultiSet const& _funcSet, F
 		);
 }
 
+std::set<FunctionDefinition const*> const& ContractLevelChecker::nonOverriddenBaseFunctions(ContractDefinition const* _contract) const
+{
+	if (!m_nonOverriddenBaseFunctions.count(_contract))
+	{
+		std::set<FunctionDefinition const*> nonOverriddenBaseFunctionSet;
+		std::set<FunctionDefinition const*> overriddenInBases;
+
+		auto const& directBases = resolveDirectBaseContracts(*_contract);
```

Comment:
It does not return by ref, so better to not use `const&`. Using `const&` might lead reviewers to think that the function returns a reference to an element in a map which could then be invalidated by recursive calls to `nonOverriddenBaseFunctions`.